datafusion/common/src/dfschema.rs

```diff
@@ -573,21 +580,21 @@ impl ExprSchema for DFSchema {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct DFField {
     /// Optional qualifier (usually a table or relation name)
-    qualifier: Option<String>,
+    qualifier: Option<OwnedTableReference>,
     /// Arrow field definition
     field: Field,
 }
 
 impl DFField {
     /// Creates a new `DFField`
-    pub fn new(
-        qualifier: Option<&str>,
+    pub fn new<R: Into<OwnedTableReference>>(
+        qualifier: Option<R>,
```

Comment:
originally had this way to try ease the breaking change, since would still support Option<&str>, but can see will cause pain if trying to pass in just None itself. maybe its better to just have this as Option<OwnedTableReference> and have the callers call into() (or maintain old method and create a new one?)

Reply:
An issue with this approach is that there are still use cases for an Option<...> parameter, such as when the caller already possesses an `Option<OwnedTableReference>` (such as the `relation` field of `Column`)

This would force them to destructure that Option manually to call the correct version, `new(...)` or `new_unqualified(...)`, which makes it more verbose for the caller

I suppose could add yet another version of new... I'm not sure I like that approach either though

Reply:
In the end I opted to leave `new(...)` as is with the `Option<...>` parameter, and introduced a new `new_qualified(...)` to allow specifying without qualifier, without needing to turbofish for a None case